lib/src/components/common_components.dart

```diff
@@ -0,0 +1,87 @@
+import 'package:flutter/foundation.dart';
+import 'package:flutter/material.dart';
+
+import '../extensions.dart';
+
+class CalendarPageHeader extends StatelessWidget {
+  /// When user taps on right arrow.
+  final VoidCallback onNextDay;
+
+  /// When user taps on left arrow.
+  final VoidCallback onPreviousDay;
+
+  /// When user taps on title.
+  final AsyncCallback onTitleTapped;
+
+  /// Date of month/day.
+  final DateTime date;
+
+  /// Provides string to display as title.
+  final StringProvider dateStringBuilder;
+
+  /// Common header for month and day view In this header user can define format
+  /// in which date will be displayed by providing [dateStringBuilder] function.
+  ///
+  const CalendarPageHeader({
```

Comment:
Don't forget to mark things as `required`

Reply:
All the required parameters are marked as required. other callbacks are not required.